osquery/events/darwin/openbsm.cpp

```diff
@@ -16,10 +18,92 @@
 
 namespace osquery {
 
+const int kQLimit = 512;
+
 DECLARE_bool(disable_audit);
+DECLARE_bool(audit_allow_process_events);
+DECLARE_bool(audit_allow_sockets);
+DECLARE_bool(audit_allow_user_events);
+DECLARE_bool(audit_allow_fim_events);
 
 REGISTER(OpenBSMEventPublisher, "event_publisher", "openbsm");
 
+Status OpenBSMEventPublisher::configureAuditPipe(FILE* au_pipe) {
+  auto au_fd = fileno(au_pipe);
+  int pr_sel_mode = AUDITPIPE_PRESELECT_MODE_LOCAL;
+
+  audit_pipe_ = nullptr;
+
+  if (ioctl(au_fd, AUDITPIPE_SET_PRESELECT_MODE, &pr_sel_mode) == -1) {
+    LOG(WARNING) << "The auditpipe:ioctl AUDITPIPE_SET_PRESELECT_MODE failed";
+    fclose(au_pipe);
+    return Status(1, "Failed to set AUDITPIPE_SET_PRESELECT_MODE");
+  }
+
+  if (ioctl(au_fd, AUDITPIPE_SET_QLIMIT, &kQLimit) == -1) {
+    LOG(INFO) << "The auditpipe:ioctl AUDITPIPE_SET_QLIMIT failed";
+  }
+
+  struct au_class_ent* ace;
```

Comment:
nitpick, initialize to `nullptr`

Reply:
I think its declared too far way from its use, if I move it closer to its use then I think it will not required to be initialized ? :)

 while ((ace = getauclassent()) != nullptr) {

Reply:
Ah yes, it might be easier to read if it’s declared closer. I think initializing it to nullptr helps with various static analysis tools. It’s something we’ve try to be consistent on throughout the codebase.

Reply:
sorry, I forgot to address that.